src/foam/u2/wizard/StepWizardletController.js

```diff
@@ -178,6 +216,30 @@ foam.CLASS({
         });
       },
     },
+    function countAvailableSections(wizardletIndex) {
+      return this.sectionAvailableSlots[wizardletIndex].reduce(
+        (total, sectionAvailable$) =>
+          sectionAvailable$.get() ? total + 1 : total,
+        0
+      );
+    },
+    function canSkipTo(wizardletIndex) {
+      let currentWizardletIndex =
+        this.screenIndexToSection(this.subStack.pos)[0];
+      for ( let w = currentWizardletIndex ; w <= wizardletIndex ; w++ ) {
+        if ( ! this.wizardlets[w].readyToSubmit() ) return false;
```

Comment:
Where is readyToSubmit set?  Seems like this is just 'validated'?

Reply:
I did this with the idea that an implementor of Wizardlet can choose what it means for something to be ready to submit. Actually this could even come up soon if we want to let the user click `next` for incomplete data but not for invalid values (both of which wouldn't pass the validation predicate)

Reply:
So this is a working concept for something we don't need and may use.

Eric is correct - this has come up in the next action, however we've decided to not do anything for next - allow no validation check, which became the necessity of https://github.com/foam-framework/foam2/pull/3560/files

@jlhughes please let us know your thoughts - I'm ok with having this level of complexity for potential future use. AND Joel - nice catch, thank you!

Reply:
'readyToSubmit' will morph into lots of custom code.  'validate' is our standard on models.   The wizard steps don't have validate but I would like to stick to the calling convention.   Open to debate.

Reply:
Unable to test any solutions because of an unrelated regression. Adding `[DNM]` tag and investigating now. :/

Reply:
nvm I just forgot to rename a function in a place after addressing another PR comment

Reply:
Moved this to `validate` and added a `mustBeValid` property - so if validation is needed before clicking next that will be set to `true`.